providers/gitlab.go

```diff
@@ -15,11 +18,49 @@ import (
 type GitLabProvider struct {
 	*ProviderData
 
-	Groups               []string
-	Verifier             *oidc.IDTokenVerifier
+	Groups       []string
+	Projects     []*GitlabProject
+	EmailDomains []string
+
 	AllowUnverifiedEmail bool
 }
 
+// GitlabProject represents a Gitlab project constraint entity
+type GitlabProject struct {
+	Name        string
+	AccessLevel int
+}
+
+// Creates a new GitlabProject struct from project string formatted as namespace/project=accesslevel
+// if no accesslevel provided, use the default one
+func gitlabProjectFromString(project string) (*GitlabProject, error) {
+	// default access level is 20
+	defaultAccessLevel := 20
+	// see https://docs.gitlab.com/ee/api/members.html#valid-access-levels
+	validAccessLevel := [4]int{10, 20, 30, 40}
+
+	parts := strings.Split(project, "=")
```

Comment:
Lets use `splitN` to have better built in validation over the input format we expect:

```suggestion
	parts := strings.SplitN(project, "=", 2)
```